syz-ci/manager.go

```diff
@@ -571,7 +582,24 @@ func (mgr *Manager) uploadBuild(info *BuildInfo, imageDir string) (string, error
 	if err := mgr.dash.UploadBuild(build); err != nil {
 		return "", err
 	}
-	mgr.uploadBuildArtifacts(build, imageDir)
+	// Normally we have no reason to upload build artifacts every restart - we don't rebuild
+	// kernels that often. So we only do that after we have built them. There's only one small
+	// exception - for debugging purposes it's better to upload more than to rebuild more.
+	assetUploadNeeded := mgr.debugStorage || mgr.buildUpdated
+	if mgr.storage != nil && assetUploadNeeded {
+		mgr.buildUpdated = false
+		// In almost all cases it's ok if we first report the build and then upload
+		// the assets - the probability that a crash happens and gets reported in between
+		// are very low.
+		go func() {
+			assets, err := mgr.uploadBuildAssets(build, imageDir)
+			if err != nil {
+				mgr.Errorf("failed to upload build assets: %v", err)
+				return
+			}
+			mgr.storage.ReportBuildAssets(build, assets...)
```

Comment:
Now that the dashapi.Build object includes Assets field, why don't we upload assets first and then in the Build object? This would exclude the whole issue of having inconsistent builds w/o assets.
Currently if syz-ci crashes, exits, restarted, VM crashes before the assets are uploaded, we won't have assets for this build permanently, right? We re-upload only in debug mode, but I assume it won't be generally enabled.

Reply:
Sounds reasonable, updated the PR.

I was worried about the time to upload the assets, but it turns out it's quite fast anyway and actually negligible if compared to the time needed to rebuild a kernel.